src/main/java/org/alfresco/repo/site/SiteMembership.java

```diff
@@ -23,37 +23,38 @@
  * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
  * #L%
  */
-package org.alfresco.repo.site;
-
-import org.alfresco.service.cmr.site.SiteInfo;
-import org.alfresco.service.cmr.site.SiteRole;
-
-/**
- * Conveys information for a member of a site.
- * 
- * @author steveglover
- *
- */
+package org.alfresco.repo.site;
+
+import org.alfresco.service.cmr.site.SiteInfo;
+
+/**
+ * Conveys information for a member of a site.
+ * 
+ * @author steveglover
+ *
+ */
 public class SiteMembership
 {
     private SiteInfo siteInfo;
-    private String personId;
+    private String id;          // contains both userId and authority Id
+    private String role;
+
+    // user Properties
     private String firstName;
     private String lastName;
-    private String role;
 
-    public SiteMembership(SiteInfo siteInfo, String personId, String firstName, String lastName,
+    public SiteMembership(SiteInfo siteInfo, String id, String firstName, String lastName,
```

Comment:
I'm not sure this works if you're trying to cover both users and groups i.e. a group doesn't have a fist and last name ;-)

In our meeting the other day you had this as a base class and then a SiteUserMembership and SiteGroupMembership implementation, wouldn't that be better? This doesn't appear to be marked as a public facing interface so changing it should be OK.

Reply:
It seems public facing method (https://github.com/Alfresco/alfresco-repository/blob/ad2fd272c65b473d6b09995038548eeb126dd526/src/main/java/org/alfresco/service/cmr/site/SiteService.java#L337)
What if some one used inside the call back?

Am i wrong?

Any way we have exposed the new methods (https://github.com/Alfresco/alfresco-repository/blob/ad2fd272c65b473d6b09995038548eeb126dd526/src/main/java/org/alfresco/service/cmr/site/SiteService.java#L588-L613)

please have a look

Reply:
Yes, you're right this object is exposed via the SiteService interface which is a public API so really this object should also be tagged with @AlfrescoPublicApi.